diff -Nrup a/bin/autoconf.as b/bin/autoconf.as
--- a/bin/autoconf.as	2021-01-28 21:46:48.000000000 +0100
+++ b/bin/autoconf.as	2023-12-20 15:30:43.096807040 +0100
@@ -89,7 +89,7 @@ exit_missing_arg='
 # restore font-lock: '
 
 # Variables.
-: ${AUTOM4TE='@bindir@/@autom4te-name@'}
+: ${AUTOM4TE='@autom4te-name@'}
 : ${trailer_m4='@pkgdatadir@/autoconf/trailer.m4'}
 autom4te_options=
 outfile=
diff -Nrup a/bin/autoheader.in b/bin/autoheader.in
--- a/bin/autoheader.in	2021-01-28 21:46:48.000000000 +0100
+++ b/bin/autoheader.in	2023-12-20 15:32:06.768522864 +0100
@@ -32,7 +32,10 @@ use warnings FATAL => 'all';
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, "$pkgdatadir";
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
@@ -54,7 +57,8 @@ use Autom4te::XFile;
 our ($config_h, %symbol, %verbatim);
 
 # Lib files.
-my $autom4te = $ENV{'AUTOM4TE'} || '@bindir@/@autom4te-name@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $autom4te = $ENV{'AUTOM4TE'} || "$prefix/bin/@autom4te-name@";
 my $config_h_in;
 my @prepend_include;
 my @include;
diff -Nrup a/bin/autom4te.in b/bin/autom4te.in
--- a/bin/autom4te.in	2021-01-28 21:46:48.000000000 +0100
+++ b/bin/autom4te.in	2023-12-20 15:32:37.921904802 +0100
@@ -28,7 +28,10 @@ use warnings FATAL => 'all';
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
@@ -48,7 +51,8 @@ use Autom4te::General;
 use Autom4te::XFile;
 
 # Data directory.
-my $pkgdatadir = $ENV{'AC_MACRODIR'} || '@pkgdatadir@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $pkgdatadir = $ENV{'AC_MACRODIR'} || "$prefix/share/autoconf";
 
 # $LANGUAGE{LANGUAGE} -- Automatic options for LANGUAGE.
 my %language;
@@ -288,7 +292,11 @@ sub load_configuration ($)
 	{
 	  fatal "$file:$.: no current language"
 	    unless defined $lang;
-	  push @{$language{$lang}}, @words;
+	  foreach (@words)
+	    {
+	      s/~/$prefix/; # make relocatable
+	      push @{$language{$lang}}, $_;
+	    }
 	}
       else
 	{
diff -Nrup a/bin/autoreconf.in b/bin/autoreconf.in
--- a/bin/autoreconf.in	2021-01-28 21:46:48.000000000 +0100
+++ b/bin/autoreconf.in	2023-12-20 15:40:00.057452368 +0100
@@ -32,10 +32,14 @@ use warnings FATAL => 'all';
 my $buildauxdir;
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
-  $buildauxdir = $ENV{'autom4te_buildauxdir'} || $pkgdatadir . '/build-aux';
+  $buildauxdir = $ENV{'autom4te_buildauxdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf', 'build-aux');
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
   # that can handle redirection and quote arguments correctly,
@@ -117,9 +121,10 @@ Written by David J. MacKenzie and Akim D
 ";
 
 # Lib files.
-my $autoconf    = $ENV{'AUTOCONF'}    || '@bindir@/@autoconf-name@';
-my $autoheader  = $ENV{'AUTOHEADER'}  || '@bindir@/@autoheader-name@';
-my $autom4te    = $ENV{'AUTOM4TE'}    || '@bindir@/@autom4te-name@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $autoconf    = $ENV{'AUTOCONF'}    || "$prefix/bin/@autoconf-name@";
+my $autoheader  = $ENV{'AUTOHEADER'}  || "$prefix/bin/@autoheader-name@";
+my $autom4te    = $ENV{'AUTOM4TE'}    || "$prefix/bin/@autom4te-name@";
 my $automake    = $ENV{'AUTOMAKE'}    || 'automake';
 my $aclocal     = $ENV{'ACLOCAL'}     || 'aclocal';
 my $libtoolize  = $ENV{'LIBTOOLIZE'}  || 'libtoolize';
diff -Nrup a/bin/autoscan.in b/bin/autoscan.in
--- a/bin/autoscan.in	2021-01-28 21:46:48.000000000 +0100
+++ b/bin/autoscan.in	2023-12-20 13:33:47.676863603 +0100
@@ -30,7 +30,10 @@ use warnings FATAL => 'all';
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
@@ -95,10 +98,11 @@ my %needed_macros =
 my $log;
 
 # Autoconf and lib files.
-my $autom4te = $ENV{'AUTOM4TE'} || '@bindir@/@autom4te-name@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $autom4te = $ENV{'AUTOM4TE'} || "$prefix/bin/@autom4te-name@";
 my $autoconf = "$autom4te --language=autoconf";
 my @prepend_include;
-my @include = ('@pkgdatadir@');
+my @include = ("$prefix/share/autoconf");
 
 # $help
 # -----
diff -Nrup a/bin/autoupdate.in b/bin/autoupdate.in
--- a/bin/autoupdate.in	2021-01-28 21:46:48.000000000 +0100
+++ b/bin/autoupdate.in	2023-12-20 15:41:17.480866597 +0100
@@ -31,7 +31,10 @@ use warnings FATAL => 'all';
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
@@ -51,14 +54,15 @@ use Autom4te::General;
 use Autom4te::XFile;
 
 # Lib files.
-my $autom4te = $ENV{'AUTOM4TE'} || '@bindir@/@autom4te-name@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $autom4te = $ENV{'AUTOM4TE'} || "$prefix/bin/@autom4te-name@";
 my $autoconf = "$autom4te --language=autoconf";
 # We need to find m4sugar.
 my @prepend_include;
-my @include = ('@pkgdatadir@');
+my @include = ("$prefix/share/autoconf");
 my $force = 0;
 # m4.
-my $m4 = $ENV{"M4"} || '@M4@';
+my $m4 = $ENV{"M4"} || 'm4';
 
 
 # $HELP
diff -Nrup a/bin/ifnames.in b/bin/ifnames.in
--- a/bin/ifnames.in	2021-01-28 21:46:48.000000000 +0100
+++ b/bin/ifnames.in	2023-12-20 13:33:47.680863791 +0100
@@ -35,7 +35,10 @@ use warnings FATAL => 'all';
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
diff -Nrup a/lib/autom4te.in b/lib/autom4te.in
--- a/lib/autom4te.in	2021-01-28 21:46:48.000000000 +0100
+++ b/lib/autom4te.in	2023-12-20 15:42:03.398890661 +0100
@@ -115,7 +115,7 @@ end-language: "Autoreconf-preselections"
 # This intermediate language is used by aclocal to build aclocal.m4.
 
 begin-language: "Autoconf-without-aclocal-m4"
-args: --prepend-include '@pkgdatadir@'
+args: --prepend-include '~/share/autoconf'
 args: --cache=autom4te.cache
 args: autoconf/autoconf.m4f
 args: acsite.m4?
@@ -142,7 +142,7 @@ end-language: "Autoconf"
 ## -------- ##
 
 begin-language: "Autotest"
-args: --prepend-include '@pkgdatadir@'
+args: --prepend-include '~/share/autoconf'
 args: autotest/autotest.m4f
 args: package.m4?
 args: local.at?
@@ -156,7 +156,7 @@ end-language: "Autotest"
 ## ---- ##
 
 begin-language: "M4sh"
-args: --prepend-include '@pkgdatadir@'
+args: --prepend-include '~/share/autoconf'
 args: m4sugar/m4sh.m4f
 args: --mode 777
 args: --language M4sugar
@@ -168,6 +168,6 @@ end-language: "M4sh"
 ## ------- ##
 
 begin-language: "M4sugar"
-args: --prepend-include '@pkgdatadir@'
+args: --prepend-include '~/share/autoconf'
 args: m4sugar/m4sugar.m4f
 end-language: "M4sugar"
